### PR TITLE
Absolute Path Arguments

### PR DESCRIPTION
**Pull request by nprisbrey**
Add support to arguments allowing the passing of alternate absolute paths instead of the default paths for weights, models, data and TensorBoard logs.

Also add some touch ups to the code introduced in PR #17 that I saw convenient. Let me know thoughts about if these changes are appropriate or not.

### CODE REVIEW

**Comment by JayOrten** [comment]
Sorry Nathan, some of these changes are probably going to conflict significantly with the changes we made in the other PR, specifically with paths, just because we were still changing some things with the download_data script.

**Comment by nprisbrey** [comment]
No worries @JayOrten, that's the magic of git! :slightly_smiling_face:

File: train_model.py [hunk 1]
```diff
@@ -324,21 +332,45 @@ def get_params(self) -> dict:
...
+        weights_folder = Path(args.weights_dir)
+    weights_folder.mkdir(parents=True, exist_ok=False)
+    print(f"Saving weight files in {weights_folder}")
+
+    # Create SummaryWriter to record logs for TensorBoard
+    if args.tboard_dir is None:
+        writer = SummaryWriter(log_dir=repo_root_dir / "logs" / model_label)
+    else:
+        writer = SummaryWriter(log_dir=f"{args.tboard_dir}/logs/{model_label}")
+    print(f"Saving TensorBoard logs in {args.tboard_dir}/logs/{model_label}")
```
Comment (KimballNJardine) [inline train_model.py]: This print statement fails to account for if args.tboard_dir is None. Please fix it to account for that.

Comment (KimballNJardine) [reply train_model.py]: Please compare this line to lines 362 and 364 for exactly what I am getting at.

**Review by KimballNJardine — CHANGES_REQUESTED** [review]
Generally good, not comment on print statement.

File: train_model.py (hunk 1 above)
Comment (nprisbrey) [reply train_model.py]: Well said @KimballNJardine! This has been addressed in [584e9dd](https://github.com/DRAGNLabs/301r_retnet/pull/21/commits/584e9dd8dc26b0ba4c7c794aa523d26c693484aa).

File: train_model.py [hunk 2]
```diff
@@ -324,21 +332,46 @@ def get_params(self) -> dict:
...
+    if args.weights_dir is None:
+        weights_folder = root_save_folder / "weights"
+    else:
+        weights_folder = Path(args.weights_dir)
+    weights_folder.mkdir(parents=True, exist_ok=False)
+    print(f"Saving weight files in {weights_folder}")
+
+    # Create SummaryWriter to record logs for TensorBoard
+    if args.tboard_dir is None:
+        tboard_log_dir = repo_root_dir / "logs" / model_label
```
Comment (KimballNJardine) [inline train_model.py]: can this be renamed tboard_log_folder to follow the pattern above? Namely,  data_folder, root_save_folder, and weights_folder?

**Review by KimballNJardine — CHANGES_REQUESTED** [review]
Minor comment on variable name

File: train_model.py (hunk 2 above)
Comment (nprisbrey) [reply train_model.py]: Great idea -- see [ec0abab](https://github.com/DRAGNLabs/301r_retnet/pull/21/commits/ec0abab52107cebfb5e75d2a7f8ff93a3b654727).

**Review by KimballNJardine — APPROVED** [review]
Approved.

**Pull request merged**